src/OrbitGl/TrackManager.cpp

```diff
@@ -342,9 +353,7 @@ void TrackManager::AddFrameTrack(const std::shared_ptr<FrameTrack>& frame_track)
 
 void TrackManager::RemoveFrameTrack(uint64_t function_id) {
   std::lock_guard<std::recursive_mutex> lock(mutex_);
-  sorted_tracks_.erase(
-      std::remove(sorted_tracks_.begin(), sorted_tracks_.end(), frame_tracks_[function_id].get()),
-      sorted_tracks_.end());
+  deleted_tracks_.push_back(frame_tracks_[function_id]);
```

Comment:
What would happen if the track was also removed from `sorted_tracks_` and `visible_tracks_` directly here? I feel like it would keep the flow simpler.

Reply:
Yeah, we've been discussing this, and I may change it accordingly. We initially had this separation of sorting & visibility update due to multithreading issues IIRC, and I didn't want to change the flow. But this should no longer be an issue, so we might as well remove it here right away.

Reply:
So we decided to keep it like this for now as it fixes the issue without changing the functionality. But this should be changed once we fix the multithreading issues with track creation.